File: CS/Extensions/DependencyObjectHelper.cs
using System.Windows;

namespace DXGridSample.Extensions {
    public class DependencyObjectHelper {
        public static object GetDataContext(DependencyObject obj) {
            return obj.GetValue(DataContextProperty);
        }
        public static void SetDataContext(DependencyObject obj, object value) {
            obj.SetValue(DataContextProperty, value);
        }
        public static readonly DependencyProperty DataContextProperty =
            DependencyProperty.RegisterAttached("DataContext", typeof(object), typeof(DependencyObjectHelper), new PropertyMetadata(null));
    }
}

File: VB/Extensions/DependencyObjectHelper.vb
Imports System.Windows

Namespace DXGridSample.Extensions

    Public Class DependencyObjectHelper

        Public Shared Function GetDataContext(ByVal obj As DependencyObject) As Object
            Return obj.GetValue(DataContextProperty)
        End Function

        Public Shared Sub SetDataContext(ByVal obj As DependencyObject, ByVal value As Object)
            obj.SetValue(DataContextProperty, value)
        End Sub

        Public Shared ReadOnly DataContextProperty As DependencyProperty = DependencyProperty.RegisterAttached("DataContext", GetType(Object), GetType(DependencyObjectHelper), New PropertyMetadata(CType(Nothing, PropertyChangedCallback)))
    End Class
End Namespace
